File: CS/Scaffolding.ManyToMany/SchoolContextDataModel/ISchoolContextUnitOfWork.cs
using System;
using System.Linq;
using System.Data;
using System.Linq.Expressions;
using System.Collections.Generic;
using Scaffolding.ManyToMany.Common.Utils;
using Scaffolding.ManyToMany.Common.DataModel;
using Scaffolding.ManyToMany.Common.DataModel.DesignTime;
using Scaffolding.ManyToMany.Common.DataModel.EntityFramework;
using Scaffolding.ManyToMany.Model;

namespace Scaffolding.ManyToMany.SchoolContextDataModel {
    /// <summary>
    /// ISchoolContextUnitOfWork extends the IUnitOfWork interface with repositories representing specific entities.
    /// </summary>
    public interface ISchoolContextUnitOfWork : IUnitOfWork {

        /// <summary>
        /// The Course entities repository.
        /// </summary>
        IRepository<Course, int> Courses { get; }

        /// <summary>
        /// The Student entities repository.
        /// </summary>
        IRepository<Student, int> Students { get; }
    }
}

File: VB/Scaffolding.ManyToMany/SchoolContextDataModel/ISchoolContextUnitOfWork.vb
Imports System
Imports System.Linq
Imports System.Data
Imports System.Data.Entity
Imports System.Linq.Expressions
Imports System.Collections.Generic
Imports Scaffolding.ManyToMany.Common.Utils
Imports Scaffolding.ManyToMany.Common.DataModel
Imports Scaffolding.ManyToMany.Common.DataModel.EntityFramework
Imports Scaffolding.ManyToMany.Model

Namespace Scaffolding.ManyToMany.SchoolContextDataModel
    ''' <summary>
    ''' ISchoolContextUnitOfWork extends the IUnitOfWork interface with repositories representing specific entities.
    ''' </summary>
    Public Interface ISchoolContextUnitOfWork
        Inherits IUnitOfWork

        ''' <summary>
        ''' The Course entities repository.
        ''' </summary>
        ReadOnly Property Courses() As IRepository(Of Course, Integer)

        ''' <summary>
        ''' The Student entities repository.
        ''' </summary>
        ReadOnly Property Students() As IRepository(Of Student, Integer)
    End Interface
End Namespace
